### ISSUE TITLE
Don't implicitly associate related objects to sqlalchemy session

### ISSUE THREAD

**Issue by mvdbeek** [galaxyproject/galaxy#13494]
https://github.com/galaxyproject/galaxy/pull/13458/ is ok as a workaround for now, but we have little control now in the calling context. We've had problems in the past with objects getting pulled into a session and getting persisted into the database when we didn't want for that to happen. The calling context should decide what gets added to the session.

**Comment by ic4f** [comment]
Absolutely. So , as per #13458, "Once we move to 2.0 and verify that the system is stable, we can remove such fixes, explicitly adding objects to the session (via session.add()) in the right place and at the right time, when needed." 

In other words, at that stage we will be *removing* explicitly added associated objects from the session, one step at a time.

**Comment by ic4f** [comment]
Removing milestone, as this is contingent on SQLAlchemy 2.0
